kitsune/search/api.py

```diff
@@ -32,6 +32,15 @@ def suggest(request):
         errors['q'] = 'This field is required.'
     if product is not None and not Product.objects.filter(slug=product).exists():
         errors['product'] = 'Could not find product with slug "{0}".'.format(product)
+    if locale not in settings.SUMO_LANGUAGES:
+        if settings.NON_SUPPORTED_LOCALES.get(locale):
+            errors['locale'] = (
+                'Locale "{0}" is not supported, but has fallback locale "{1}".'.format(
+                    locale, settings.NON_SUPPORTED_LOCALES[locale])
+            )
```

Comment:
There are a lot of locales in NON_SUPPORTED_LOCALES with a value of None, which means they redirect to WIKI_DEFAULT_LANGUAGE. For consistency the if statement should probably be `if locale in settings.NON_SUPPORTED_LOCALES` and the message should use `settings.NON_SUPPORTED_LOCALES.get(locale, 'en-US')`.